promql/engine.go

```diff
@@ -559,6 +546,21 @@ func (ng *Engine) populateIterators(ctx context.Context, s *EvalStmt) (storage.Q
 	return querier, err
 }
 
+// extractFuncFromPath walks up the path and searches for the first instance of
```

Comment:
I think we should only look back one node. For example `max(1-metric)` might in fact be a `min` in practice, so a hint of `max` would incorrect.

Reply:
Mh, maybe early exit on binary expressions? Because just looking back one node won't work for `max_over_time(a[5m])` or `count((a))` as the range selector or parens are their own nodes.

Reply:
SGTM